buildSrc/src/main/kotlin/io/spine/internal/gradle/github/pages/UpdateGitHubPages.kt

```diff
@@ -0,0 +1,403 @@
+        Cli(repoBaseDir).execute("git", "checkout", gitHubPagesBranch)
+    }
+
+    /**
+     * Creates an SSH key with the credentials from [gitHubAccessKey]
+     * and registers it by invoking the `register-ssh-key.sh` script.
+     */
+    private fun registerSshKey(gitHubAccessKey: File) {
+        val sshConfigFile = File("${System.getProperty("user.home")}/.ssh/config")
+        if (!sshConfigFile.exists()) {
+            val parentDir = sshConfigFile.canonicalFile.parentFile
+            parentDir.mkdirs()
+            sshConfigFile.createNewFile()
+        }
+        sshConfigFile.appendText(
+            lineSeparator() +
+                    "Host github.com-publish" + lineSeparator() +
+                    "User git" + lineSeparator() +
+                    "IdentityFile ${gitHubAccessKey.absolutePath}" + lineSeparator()
+        )
+
+        Cli(rootFolder).execute(
+            "${rootFolder.absolutePath}/config/scripts/register-ssh-key.sh",
+            gitHubAccessKey.absolutePath
+        )
+    }
+
+    /**
+     * Reads `REPO_SLUG` environment variable and returns its value.
+     *
+     * In case it is not set, a [GradleException] is thrown.
+     */
+    private fun repoSlug(): String {
+        val repoSlug = System.getenv(repoSlugVar)
+        if (repoSlug == null || repoSlug.isEmpty()) {
+            throw GradleException("`REPO_SLUG` environment variable is not set.")
+        }
+        return repoSlug
+    }
+
+    /**
+     * Returns the GitHub URL to the project repository.
+     *
+     * <p>A CI instance comes with an RSA key. However, of course, the default key has no
+     * privileges in Spine repositories. Thus, we add our own RSA key — `deploy_rsa_key`.
+     * It must have write rights in the associated repository. Also, we don't want that key
+     * to be used for anything else but GitHub Pages publishing.
+     * Thus, we configure the SSH agent to use the `deploy_rsa_key`
+     * only for specific references, namely in `github.com-publish`.
+     */
+    private fun gitHost(repoSlug: String): String {
+        return "git@github.com-publish:${repoSlug}.git"
+    }
+
+    /**
+     * Locates `deploy_key_rsa` in the passed [rootFolder] and returns it as a [File].
+     *
+     * If it is not found, a [GradleException] is thrown.
+     */
+    private fun gitHubKey(rootFolder: File): File {
+        val gitHubAccessKey = File("${rootFolder.absolutePath}/deploy_key_rsa")
+
+        if (!gitHubAccessKey.exists()) {
+            throw GradleException(
+                "File $gitHubAccessKey does not exist. It should be encrypted" +
+                        " in the repository and decrypted on CI."
+            )
+        }
+        return gitHubAccessKey
+    }
+
+    private fun isSnapshot(version: String): Boolean {
+        return version.contains("snapshot", true)
+    }
+}
+
+/**
+ * The extension for configuring the `UpdateGitHubPages` plugin.
+ */
+class UpdateGitHubPagesExtension
```

Comment:
Please pull the extension to the package level.

Reply:
Done.